webapp/server/methods.js

```diff
@@ -335,4 +335,36 @@ Meteor.methods({
       }
     });
   },
+
+  // Get the genes that should have icons appear next to their names.
+  // They are contained in a special geneSetCollection named
+  // "GeneSets appearing as icons. Do not delete. oRZvz3Gbim"
+  // To specify the icon & color, each gene_set's name should be, eg, "yellow star".
+  // (this is a meteor method & not a publication so we can transform the find server-side)
+  getGeneInfos(){
+    let findBlessedSet = {"name":"GeneSets appearing as icons. Do not delete. oRZvz3Gbim"};
+
+    let addExtraInfo = function(geneSet) {
```

Comment:
Yeah, using `this.added`. The other option would be to send the gene set to
the client in a publication and handle the transformation there... This way
works fine!

On Thu, Aug 11, 2016 at 3:44 PM, e-t-k <notifications@github.com> wrote:

> In webapp/server/methods.js
> <https://github.com/UCSC-MedBook/patient-care/pull/75#discussion_r74517499>
> :
>
> > @@ -335,4 +335,36 @@ Meteor.methods({
> >        }
> >      });
> >    },
> > +
> > +  // Get the genes that should have icons appear next to their names.
> > +  // They are contained in a special geneSetCollection named
> > +  // "GeneSets appearing as icons. Do not delete. oRZvz3Gbim"
> > +  // To specify the icon & color, each gene_set's name should be, eg, "yellow star".
> > +  // (this is a meteor method & not a publication so we can transform the find server-side)
> > +  getGeneInfos(){
> > +    let findBlessedSet = {"name":"GeneSets appearing as icons. Do not delete. oRZvz3Gbim"};
> > +
> > +    let addExtraInfo = function(geneSet) {
>
> How? Using the low-level interface in http://docs.meteor.com/api/
> pubsub.html#Meteor-publish ? I can't find docs for it now but my
> understanding is that you can't use transform on server-side
> publications, only client-side.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/UCSC-MedBook/patient-care/pull/75/files/ed36f6507081438ad06ae3d6622fbb63af50a89f#r74517499>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABRIv0jA2p2ax6aKPg66TM7wCsOiv4TKks5qe6XcgaJpZM4JgmBX>
> .
>